var/spack/repos/builtin/packages/hypre-cmake/package.py

```diff
@@ -92,38 +94,36 @@ def setup_build_environment(self, env):
 
     @run_after("install")
     def cache_test_sources(self):
-        self.cache_extra_test_sources(self.extra_install_tests)
+        cache_extra_test_sources(self, self.extra_install_tests)
 
     @property
     def _cached_tests_work_dir(self):
         """The working directory for cached test sources."""
         return join_path(self.test_suite.current_test_cache_dir, self.extra_install_tests)
 
-    def test(self):
-        """Perform smoke test on installed HYPRE package."""
+    def test_hypre_cmake(self):
+        """Perform smoke tests on installed HYPRE package."""
         if "+mpi" not in self.spec:
-            print("Skipping: HYPRE must be installed with +mpi to run tests")
-            return
+            raise SkipTest("Package must be installed with +mpi to run tests")
 
         # Build copied and cached test examples
-        self.run_test(
-            "make",
-            ["HYPRE_DIR={0}".format(self.prefix), "bigint"],
-            purpose="test: building selected examples",
-            work_dir=self._cached_tests_work_dir,
-        )
-
-        # Run the examples built above
-        for exe in ["./ex5big", "./ex15big"]:
-            self.run_test(
-                exe,
-                [],
-                [],
-                installed=False,
-                purpose="test: ensuring {0} runs".format(exe),
-                skip_missing=True,
-                work_dir=self._cached_tests_work_dir,
-            )
+        with working_dir(self._cached_tests_work_dir):
+            make = which("make")
+            make(f"HYPRE_DIR={self.prefix}", "bigint")
+
+            for exe_name in ["ex5big", "ex15big"]:
+                with test_part(
+                    self, f"test_hypre_cmake_{exe_name}", purpose=f"Ensure {exe_name} runs"
```

Comment:
To match more descriptive name:

```suggestion
                    self, f"test_bigint_{exe_name}", purpose=f"Ensure {exe_name} runs"
```